tools/src/main/java/org/apache/kafka/tools/ReplicaVerificationTool.java

```diff
@@ -0,0 +1,744 @@
+                    .collect(Collectors.groupingBy(
+                        TopicPartitionReplica::brokerId,
+                        Collectors.mapping(
+                            replica -> new TopicPartition(replica.topic(), replica.partition()),
+                            Collectors.toList()
+                        )
+                    ));
+                LOG.debug("Topic partitions per broker: {}", brokerToTopicPartitions);
+
+                Map<TopicPartition, Integer> expectedReplicasPerTopicPartition = topicPartitionReplicas.stream()
+                    .collect(Collectors.groupingBy(
+                        replica -> new TopicPartition(replica.topic(), replica.partition()),
+                        Collectors.collectingAndThen(
+                            Collectors.toList(),
+                            List::size
+                        )
+                    ));
+                LOG.debug("Expected replicas per topic partition: {}", expectedReplicasPerTopicPartition);
+
+                List<TopicPartition> topicPartitions = filteredTopicMetadata.stream()
+                    .flatMap(topicMetadata -> topicMetadata.partitions().stream()
+                        .map(partitionMetadata -> new TopicPartition(topicMetadata.name(), partitionMetadata.partition()))
+                    )
+                    .collect(Collectors.toList());
+
+                Properties consumerProps = consumerConfig(brokerList);
+
+                ReplicaBuffer replicaBuffer = new ReplicaBuffer(expectedReplicasPerTopicPartition,
+                    initialOffsets(topicPartitions, consumerProps, options.initialOffsetTime()),
+                    brokerToTopicPartitions.size(), options.reportInterval());
+
+                // create all replica fetcher threads
+                int verificationBrokerId = brokerToTopicPartitions.entrySet().iterator().next().getKey();
+                AtomicInteger counter = new AtomicInteger(0);
+                List<ReplicaFetcher> fetcherThreads = brokerToTopicPartitions.entrySet().stream()
+                    .map(entry -> {
+                        int brokerId = entry.getKey();
+                        Iterable<TopicPartition> partitions = entry.getValue();
+                        return new ReplicaFetcher(
+                            "ReplicaFetcher-" + brokerId,
+                            brokerInfo.get(brokerId),
+                            partitions,
+                            topicIds,
+                            replicaBuffer,
+                            options.fetchSize(),
+                            options.maxWaitMs(),
+                            1,
+                            brokerId == verificationBrokerId,
+                            consumerProps,
+                            counter.incrementAndGet()
+                        );
+                    })
+                    .collect(Collectors.toList());
+
+                Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
I think we can keep the shutdown hook thread name with something like:
```
Runtime.getRuntime().addShutdownHook(new Thread(() -> {
   ...
}, "ReplicaVerificationToolShutdownHook"));

Reply:
sure